src/mlpack/methods/preprocess/preprocess_string_main.cpp

```diff
@@ -0,0 +1,267 @@
+    "\n\n"
+    "So, a simple example where we want to preprocess string the dataset " +
+    PRINT_DATASET("X") + ", which is having string data in its 3 Column."
+    "\n\n" +
+    PRINT_CALL("preprocess_string", "actual_dataset", "X",
+        "preprocess_dataset", "X", "lowercase", 1, "stopwords",
+        1, "punctuation", 1, "dimension", 3) +
+    "\n\n",
+    SEE_ALSO("@preprocess_binarize", "#preprocess_binarize"),
+    SEE_ALSO("@preprocess_describe", "#preprocess_describe"),
+    SEE_ALSO("@preprocess_imputer", "#preprocess_imputer"));
+
+PARAM_STRING_IN_REQ("actual_dataset", "File containing the reference dataset.",
+    "t");
+PARAM_STRING_IN_REQ("preprocess_dataset", "File containing the preprocess "
+    "dataset.", "o");
+PARAM_STRING_IN("column_delimiter", "delimeter used to seperate Column in files"
+    "example '\\t' for '.tsv' and ',' for '.csv'.", "d", "\t");
+PARAM_STRING_IN("delimiter", "A set of chars that is used as delimeter to"
+    "tokenize the string dataset ", "D", " ");
+PARAM_STRING_IN("stopwordsfile", "File containing stopwords", "S", "");
+
+PARAM_FLAG("lowercase", "convert to lowercase.", "l");
+PARAM_FLAG("punctuation", "Remove punctuation.", "p");
+PARAM_FLAG("stopwords", "Remove stopwords.", "s");
+PARAM_VECTOR_IN_REQ(std::string, "dimension", "Column which contains the"
+    "string data. (1 by default)", "c");
+
+using namespace mlpack;
+using namespace mlpack::util;
+using namespace arma;
+using namespace std;
+
+static void mlpackMain()
+{
+  // Parse command line options.
+  // Extracting the filename
+  const std::string filename = CLI::GetParam<std::string>("actual_dataset");
+  // This is very dangerous, Let's add a check tommorrow.
+  std::string column_delimiter;
+  if (CLI::HasParam("column_delimiter"))
+  {
+    column_delimiter = CLI::GetParam<std::string>("column_delimiter");
+    // Allow only 3 delimiters.
+    RequireParamValue<std::string>("column_delimiter", [](std::string del)
+        { return del == "\t" || del == "," || del == " "; }, true,
+        "Delimiter should be either \\t (tab) or , (comma) or ' ' (space) ");
+  }
+  else
+  {
+      if (data::Extension(filename) == "csv")
+      {
+        column_delimiter = ",";
+        Log::Warn << "Found csv Extension, taking , as column_delimiter. \n";
+      }
+      else if (data::Extension(filename) == "tsv" ||
+          data::Extension(filename) == "txt")
+      {
+        column_delimiter = "\t";
+        Log::Warn << "Found tsv or txt Extension, taking \\t as"
+        "column_delimiter. \n";
+      }
+      else
+      {
+        Log::Warn << "column_delimiter not specified, taking default value \n";
+      }
+  }
+  // Handling Dimension vector
+  std::vector<std::string> temp_dimension =
+      CLI::GetParam<std::vector<std::string> >("dimension");
+  std::set<size_t>dimension;
+  int columnstartindex, columnendindex;
+  size_t found;
+  for (size_t i = 0; i < temp_dimension.size(); i++)
```

Comment:
I think the whole loop can be moved to a separate function that returns `std::set<size_t>`.